File: frontend/src/types/agent.ts
export type StepType = 'user' | 'thinking' | 'search' | 'memory' | 'code' | 'scrape' | 'skill' | 'result'
export type StepStatus = 'running' | 'completed' | 'error'

export interface UserData {
  content: string
  fileName?: string
}

export interface ThinkingData {
  content: string
}

export interface SearchData {
  query: string
  results: Array<{ title: string; url: string; snippet: string }>
  answer?: string
}

export interface MemoryData {
  action: 'read' | 'write'
  key: string
  value: string
}

export interface CodeData {
  language: 'javascript' | 'python' | 'cpp' | 'c' | 'java' | string
  code: string
  output: string
  status: 'running' | 'completed' | 'error'
}

export interface ScrapeData {
  url: string
  content: string
}

export interface SkillData {
  skill_name: string
  skill_icon: string
  content: string
  sub_steps: string[]
  language?: string
  render_type?: string
  progress?: number
  current_phase?: string
  progress_note?: string
  progress_indeterminate?: boolean
  progress_label?: string
  streaming_code?: boolean
  draft_resource?: {
    client_draft_id: string
    resource_type: string
    title: string
    content: unknown
    course_name?: string | null
    knowledge_points?: string[]
    kp_weights?: Record<string, number>
    course_bindings?: Array<{
      course_name: string
      knowledge_points: string[]
      weight?: number
      kp_weights?: Record<string, number>
    }>
    save_required?: boolean
  } | null
}

export interface ResultData {
  content: string
}

export interface AgentStep {
  stepId: string
  stepType: StepType
  status: StepStatus
  title: string
  agentName?: string
  data: UserData | ThinkingData | SearchData | MemoryData | CodeData | ScrapeData | SkillData | ResultData
  expanded: boolean
  timestamp: number
}

export interface AgentTask {
  id: number
  title: string
  status: 'idle' | 'running' | 'completed'
  steps: AgentStep[]
  createdAt: string
}

export interface AgentCollaborationEvent {
  type: 'agent_started' | 'agent_progress' | 'agent_completed' | 'agent_failed' | 'graph_done' | string
  event_id: string
  task_id?: string
  agent_key: string
  agent_name: string
  role: string
  stage: string
  status: 'waiting' | 'running' | 'completed' | 'error' | string
  timestamp: string
  input_summary?: string
  output_summary?: string
  resource_type?: string
  resource_id?: number
  resource_title?: string
  ppt_session?: Record<string, unknown>
  course_name?: string
  knowledge_points?: string[]
  error?: string
}

export interface StepEvent {
  type: string
  step_type: StepType
  step_id: string
  status: StepStatus
  title: string
  agent_name?: string
  data: Record<string, unknown>
}

export interface UploadedFile {
  fileName: string
  content: string
  size: number
}

File: backend/schemas/agent.py
from pydantic import BaseModel
from typing import Optional, Union


class ThinkingData(BaseModel):
    content: str = ""


class SearchResult(BaseModel):
    title: str = ""
    url: str = ""
    snippet: str = ""


class SearchData(BaseModel):
    query: str = ""
    results: list[SearchResult] = []
    answer: str = ""


class MemoryData(BaseModel):
    action: str = "read"
    key: str = ""
    value: str = ""


class CodeData(BaseModel):
    language: str = "javascript"
    code: str = ""
    output: str = ""
    status: str = "completed"


class ScrapeData(BaseModel):
    url: str = ""
    content: str = ""


class SkillData(BaseModel):
    skill_name: str = ""
    skill_icon: str = "🔧"
    content: str = ""
    sub_steps: list[str] = []
    language: str = ""


class ResultData(BaseModel):
    content: str = ""


class StepEvent(BaseModel):
    type: str = "step"
    step_type: str
    step_id: str
    status: str = "running"
    title: str = ""
    data: Optional[dict] = None


class AgentExecuteRequest(BaseModel):
    user_id: str
    task_description: str
    conversation_id: Optional[int] = None
    history: Optional[list[dict]] = None
    file_content: Optional[str] = None
    file_name: Optional[str] = None


class UploadResponse(BaseModel):
    ok: bool = True
    file_name: str = ""
    content: str = ""
    size: int = 0
    error: str = ""
